receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +383,40 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if len(partitionID) > 0 && r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), partitionID)
+		} else {
+			r.writeSuccessResponseWithoutAck(ctx, resp, ld.LogRecordCount())
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) validateAndExtractChannelHeader(req *http.Request) (string, error) {
+	// check channel header exists
+	partitionID := req.Header.Get(splunk.HTTPSplunkChannelHeader)
+
+	if len(partitionID) == 0 {
+		return "", nil
+	}
```

Comment:
can we write it like this?
```
func (r *splunkReceiver) validateAndExtractChannelHeader(req *http.Request) (string, error) {
       if !req.Header.Has(splunk.HTTPSplunkChannelHeader) {
           return "", nil
       }
       
       partitionID := req.Header.Get(splunk.HTTPSplunkChannelHeader)
       if len(partitionID) == 0 {
		return "", errors.New(responseErrInvalidDataChannel)
	}
	
       // check validity of channel
	_, err := uuid.Parse(partitionID)
	if err != nil {
		return "", errors.New(responseErrInvalidDataChannel)
	}
	return partitionID, nil
}
```

Reply:
has() is actually not exposed so Get is the only method I can call on Header unfortunately :( There's no good way to tell if a specific header exists or is empty afaik

Reply:
header is also a map, we can also write it like this:
```
val, ok := req.Header[httpContentEncodingHeader]
```